src/App/Services/MobilePasswordRepromptService.cs

```diff
@@ -42,7 +47,37 @@ private async Task<bool> ValidatePasswordAsync(string password)
                 return false;
             };
 
-            return await _cryptoService.CompareAndUpdateKeyHashAsync(password, null);
+            var email = await _stateService.GetEmailAsync();
+            var kdfConfig = await _stateService.GetActiveUserCustomDataAsync(a => new KdfConfig(a?.Profile));
+            var masterKey = await _cryptoService.MakeMasterKeyAsync(password, email, kdfConfig);
+
+            var storedPasswordHash = await _cryptoService.GetMasterKeyHashAsync();
+            if (storedPasswordHash == null)
+            {
+                var oldKey = await _secureStorageService.GetAsync<string>("oldKey");
+                if (masterKey.KeyB64 == oldKey)
+                {
+                    var localPasswordHash = await _cryptoService.HashMasterKeyAsync(password, masterKey, HashPurpose.LocalAuthorization);
+                    await _secureStorageService.RemoveAsync("oldKey");
+                    await _cryptoService.SetMasterKeyHashAsync(localPasswordHash);
```

Comment:
@mpbw2 Do you remember why we have this `oldKey`? It seems like we could probably remove support for it at this point.

Reply:
@jlf0dev It was from a migration that took place before I joined, so I agree it's probably safe to let it go.

Reply:
Awesome 👍